openmc/mgxs_library.py

```diff
@@ -553,23 +585,526 @@ def nu_fission(self, nu_fission):
                 msg = "Invalid Shape of Nu_fission!"
                 raise ValueError(msg)
         else:
-            # Get shape of nu_fission so we can figure if we need chi or not
+            # Get shape of nu_fission to determine if we need chi or not
             if nu_fission.shape == shape_vec:
                 self._use_chi = True
-                shape = shape_vec
             elif nu_fission.shape == shape_mat:
                 self._use_chi = False
-                shape = shape_mat
             else:
                 msg = "Invalid Shape of Nu_fission!"
                 raise ValueError(msg)
 
         # check we have a numpy list
-        check_type("nu_fission", nu_fission, np.ndarray, expected_iter_type=Real)
+        check_type("nu_fission", nu_fission, np.ndarray,
+                   expected_iter_type=Real)
         self._nu_fission = np.copy(nu_fission)
         if np.sum(self._nu_fission) > 0.0:
             self._fissionable = True
 
+    def set_total(self, total, **kwargs):
+        if (isinstance(total, openmc.mgxs.TotalXS) or
+            isinstance(total, openmc.mgxs.TransportXS)):
+            # Make sure passed MGXS object contains correct group structure
+            if self.energy_groups != total.energy_groups:
+                msg = 'Group structure of provided data does not match' \
+                      ' group structure of XSdata object'
+                raise ValueError(msg)
+            # Get openmc.mgxs.get_xs() arguments from kwargs
+            # nuclides, xs_type, and value will have sane defaults but can be
+            # overridden by kwards
+            if 'nuclides' in kwargs:
+                nuclides = kwargs['nuclides']
+            else:
+                nuclides = 'sum'
+            if 'xs_type' in kwargs:
+                xs_type = kwargs['xs_type']
+            else:
+                xs_type = 'macro'
+            if 'value' in kwargs:
+                value = kwargs['value']
+            else:
+                value = 'mean'
+            # subdomains is required from the kwargs as this is specific to
+            # this XSdata object.
+            if 'subdomains' in kwargs:
```

Comment:
So if you move to explicit keyword arguments, I would suggest just removing `subdomains` altogether and allowing the `MGXS.get_xs(...)` method use its default `subdomains='all'` parameter. To be robust, you might add a check to each setter that the `MGXS.domain_type` is `'universe'` or `'material'` and *not* `'distribcell'` since it simply wouldn't work within our current API.